presto-connector-tests/src/main/java/com/facebook/presto/connector/unittest/MetadataTableTest.java

```diff
@@ -0,0 +1,238 @@
+@RequiredFeatures({CREATE_TABLE_AS, DROP_TABLE})
+public interface MetadataTableTest
+        extends BaseMetadataTest
+{
+    default void testRenameTable(SchemaTableName initial, SchemaTableName renamed)
+            throws Exception
+    {
+        ConnectorSession session = new TestingConnectorSession(ImmutableList.of());
+
+        ConnectorTableMetadata tableMetadata = new ConnectorTableMetadata(
+                initial,
+                ImmutableList.of(
+                        new ColumnMetadata("bigint_column", BIGINT),
+                        new ColumnMetadata("double_column", DOUBLE)),
+                getTableProperties());
+
+        try (AutoCloseable cleanup = new AllCloser(withSchemas(session, schemaNamesOf(initial, renamed)))) {
+            run(ImmutableList.of(
+                    metadata -> {
+                        ConnectorOutputTableHandle handle = metadata.beginCreateTable(session, tableMetadata, Optional.empty());
+                        metadata.finishCreateTable(session, handle, ImmutableList.of());
+                    },
+                    metadata -> metadata.renameTable(session, metadata.getTableHandle(session, initial), renamed),
+                    metadata -> assertEquals(getOnlyElement(metadata.listTables(session, renamed.getSchemaName())), renamed),
+                    metadata -> metadata.dropTable(session, metadata.getTableHandle(session, renamed))));
+        }
+    }
+
+    @Test
+    @RequiredFeatures(RENAME_TABLE)
+    default void testRenameTableWithinSchema()
+            throws Exception
+    {
+        testRenameTable(
+                schemaTableName("initial"),
+                schemaTableName("renamed"));
+    }
+
+    @Test
+    @RequiredFeatures({CREATE_SCHEMA, DROP_SCHEMA})
+    default void testRenameTableAcrossSchema()
+            throws Exception
+    {
+        testRenameTable(
+                schemaTableName("initialSchema", "initialTable"),
+                schemaTableName("renamedSchema", "renamedTable"));
+    }
+
+    @Test
+    default void testListColumnsOneSchema()
+            throws Exception
+    {
+        SchemaTableName schemaTableName1 = schemaTableName("table1");
+        SchemaTableName schemaTableName2 = schemaTableName("table2");
+
+        ConnectorSession session = new TestingConnectorSession(ImmutableList.of());
+
+        List<ColumnMetadata> columns = ImmutableList.of(
+                new ColumnMetadata("bigint_column", BIGINT),
+                new ColumnMetadata("double_column", DOUBLE));
+
+        List<ColumnMetadata> expectedColumns = withSystemColumns(columns);
+
+        ConnectorTableMetadata tableMetadata11 = new ConnectorTableMetadata(schemaTableName1, columns, getTableProperties());
+        ConnectorTableMetadata tableMetadata12 = new ConnectorTableMetadata(schemaTableName2, columns, getTableProperties());
+
+        try (AutoCloseable cleanup = withTables(session, tableMetadata11, tableMetadata12)) {
```

Comment:
API is now commented and more consistent, subject to the aforementioned constraint on variadic functions. Let me know if you think it's nicer ;-)